Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/ServiceFactory.kt

```diff
@@ -0,0 +1,121 @@
+package de.rki.coronawarnapp.http
+
+import de.rki.coronawarnapp.CoronaWarnApplication
+import de.rki.coronawarnapp.http.service.DistributionService
+import de.rki.coronawarnapp.http.service.SubmissionService
+import de.rki.coronawarnapp.http.service.VerificationService
+import de.rki.coronawarnapp.risk.TimeVariables
+import okhttp3.Cache
+import okhttp3.ConnectionPool
+import okhttp3.ConnectionSpec
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.protobuf.ProtoConverterFactory
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+class ServiceFactory {
+    companion object {
+        /**
+         * 10 MiB
+         */
+        private const val HTTP_CACHE_SIZE = 50L * 1024L * 1024L
+
+        /**
+         * Cache file name
+         */
+        private const val HTTP_CACHE_NAME = "http_cache"
+    }
+
+    /**
+     * List of interceptors, e.g. logging
+     */
+    private val mInterceptors: List<Interceptor> = listOf(
+        HttpLoggingInterceptor().also {
```

Comment:
I'd say wrap it around `if (BuildConfig.DEBUG)`, otherwise I think that's a security vulnerability since requests will be logged during the release build and anyone can attach to Logcat to read the output.